src/main/java/net/revelc/code/impsort/ImpSort.java

```diff
@@ -85,20 +95,20 @@ public ImpSort(final Charset sourceEncoding, final Grouper grouper, final boolea
   public Result parseFile(final Path path) throws IOException {
     byte[] buf = Files.readAllBytes(path);
     if (buf.length == 0) {
-      throw new ImpSortException(path, Reason.EMPTY_FILE);
+      return Result.EMPTY_FILE;
     }
     String file = new String(buf, sourceEncoding);
     LineEnding fileLineEnding = LineEnding.determineLineEnding(file);
-    if (fileLineEnding == LineEnding.UNKNOWN) {
-      throw new ImpSortException(path, Reason.UNKNOWN_LINE_ENDING);
-    }
     LineEnding impLineEnding;
     if (lineEnding == LineEnding.KEEP) {
+      if (fileLineEnding == LineEnding.UNKNOWN) {
```

Comment:
I think it may still have been good to emit a warning here if it's `UNKNOWN` and not `KEEP`. In that case, it treats it as `AUTO`?

Reply:
If it's UNKNOWN and not KEEP, it doesn't matter what if it can be detected or not... it will just be replaced, as configured by the user. KEEP is the only situation where it matters what was there before.

Reply:
Sorry, of course, that is the "normal" behavior it always had.